### ISSUE TITLE
Dzień dobry. Chciałem się zapytać czy jest gdzieś możliwość sprawdzenia ilość aktywnych sesji bo token traci ważność szybciej niż 12h?

### ISSUE THREAD

**Issue by albertmazur** [allegro/allegro-api#10509]

**Comment by albertmazur** [comment]
Dzień dobry. Chciałem się zapytać czy jest gdzieś możliwość sprawdzenia ilość aktywnych sesji bo token traci ważność szybciej niż 12h?

**Comment by MartaNowaczyk** [comment]
Dzień dobry, nie udostępniamy nigdzie takich informacji do sprawdzenia.

**Comment by albertmazur** [comment]
A jak mogę zweryfikować że przyczyną nie ważności tokena jest przekroczona ilość sesji?

**Comment by MartaNowaczyk** [comment]
Musisz niestety zweryfikować działania po swojej stronie. Wprost informacji nie dostaniesz.

**Comment by albertmazur** [comment]
A czy możecie mi opisać dział sesja na przeglądarce czy ona jest odświeżana tak jak token 12h?

**Comment by albertmazur** [comment]
Dzień dobry. Jeśli mam konto gdzie około 15 pracowników się loguje i z tego konta korzysta inne api to jam mam rozwiązać problem sesji jeśli jest przekraczana?

**Comment by PrzemyslawLukanowski** [comment]
Dzień dobry, nie ma tutaj prostego rozwiązania poza zminimalizowaniem liczby aktywnych sesji - 20 to liczba stała, po której przekroczeniu, pierwsza sesja będzie automatycznie wygasać. Użytkownik korzystający z serwisu WWW po wykonaniu niezbędnych operacji może się wylogować z konta - w ten sposób jedna aktywna sesja zostanie zamknięta, tworząc miejsce dla pozostałych.

**Comment by albertmazur** [comment]
A czy stworzenie drugiego konta i zrobienie powiązania mogła by rozwiązać teb problem?

**Comment by PrzemyslawLukanowski** [comment]
Tak, ponieważ token będziesz generować w ramach innego użytkownika. Taki użytkownik nie będzie miał jednak za pośrednictwem API dostępu do informacji dla pierwszego konta.

**Comment by PrzemyslawLukanowski** [comment]
Prócz tych opisanych we [fragmencie poradnika](https://developer.allegro.pl/tutorials/uwierzytelnianie-i-autoryzacja-zlq9e75GdIR#kiedy-token-straci-waznosc) powodem zwrócenia tego komunikatu może być to, że np. posługujesz się refresh tokenem ze środowiska testowego na produkcji.
Możesz wysłać nam cały request, wraz z użytym refresh tokenem, przez [formularz kontaktowy](https://allegro.pl/pomoc/kontakt), zweryfikujemy to. Dopisz wtedy, że dane dotyczą zgłoszenia #10509 na GitHubie.

**Comment by albertmazur** [comment]
Ale jak wiadomości do klientów mi się wysyłała od razu po zalogowaniu tylko po jakimś czasie kiedy automatycznie próbuje sprawdzić token to raczej mogę wykluczyć że korzystam z tokena z środowiska testowego?

**Comment by PrzemyslawLukanowski** [comment]
Prawdopodobnie tak, dlatego najlepiej, jeśli dosłałbyś nam ten konkretny przypadek.

**Comment by albertmazur** [comment]
Dzień dobry, wysyłałem wiadomość przez formularz kontaktowy i nie dostałem żadnej odpowiedź

**Comment by albertmazur** [comment]
Jaki mam wybrać temat kontaktu żeby ułatwić znalezienie wiadomości?

**Comment by PrzemyslawLukanowski** [comment]
Z tego co widzę nie musisz wybierać żadnego tematu, wystarczy, że zaznaczysz "sprzedaję".

**Comment by albertmazur** [comment]
Wiadomość została wysłana 24.12.2024 a dalej nie dostałem odpowiedzi

**Comment by PrzemyslawLukanowski** [comment]
Prosiłbym o przypomnienie się w utworzonym zgłoszeniu, możesz załączyć najnowszy przypadek, gdzie otrzymujesz kod 401. W międzyczasie możesz podesłać tutaj w wątku clientId, dla którego występuje ten problem, wraz z dniem i godziną - spróbujemy chociaż na tej podstawie przenalizować ten przypadek, skoro występują problemy z przekazaniem wiadomości po stronie obsługi klienta.

**Comment by albertmazur** [comment]
Dzień dobry, przysyłam clientId 04cca6e83a88469794e2523e32f31294 błąd pojawia się codziennie około godziny 12:00

**Comment by PrzemyslawLukanowski** [comment]
Dziękuję. Sprawdziłem nasze logi, jednak nie udało mi się namierzyć tych błędów. W takim przypadku musimy poczekać na logi, które załączyłeś przez formularz - Twoje zgłoszenie powinno otrzymać odpowiedni numer i jeśli wskazałeś w nim, że dane należy przekazać do zespołu API, obsługa klienta powinna nam je wkrótce dostarczyć.

W międzyczasie pozostaje Ci tylko się upewnić:
 - czy na pewno korzystacie z adresu https://allegro.pl/auth/oauth/token i w parametrze `grant_type` przekazujecie `refresh_token`, a w parametrze `refresh_token` token odświeżający, który otrzymaliście wraz z access tokenem;
 - czy na pewno załączacie nagłówek HTTP: Authorization z autoryzacją typu Basic w postaci Base64, z zakodowanymi danymi client_id:client_secret (wraz ze znakiem “:” pomiędzy wartościami)

**Comment by PrzemyslawLukanowski** [comment]
Otrzymaliśmy Twoje dane, nie widzimy w nich jednak requestu, zaczynają się od "początek response". Kolejne informacje wskazują jednak na to, że nie wysyłasz do nas nagłówka Authorization i to powoduje wspomniany błąd.

**Comment by PrzemyslawLukanowski** [comment]
Rozumiem, jednak nie widzimy także innych danych, o których wspomniałem wyżej:

> czy na pewno korzystacie z adresu https://allegro.pl/auth/oauth/token i w parametrze grant_type przekazujecie refresh_token, a w parametrze refresh_token token odświeżający, który otrzymaliście wraz z access tokenem;

Trudno nam zlokalizować przyczynę błędu na podstawie tych danych. Patrząc na inne wątki związane z tym problemem, sugeruję usunąć `"string"` z trzeciej lini, czyli zamiast tego, co powyżej:

`'Authorization: Basic ' . base64_encode('$clientId:$clientSecret')`

**Comment by albertmazur** [comment]
Wyssałem nowy error który już zawiera w nagłówku Authorization czy możecie ją sprawdzić?

**Comment by PrzemyslawLukanowski** [comment]
Oczywiście, jak tylko dane zostaną nam przekazane, zapoznamy się z nimi.

**Comment by PrzemyslawLukanowski** [comment]
Dziękuję za informację. Jak tylko otrzymamy pełne logi, zapoznamy się z nimi i wrócimy z odpowiedzią.

**Comment by albertmazur** [comment]
Dziś jeszcze raz wyssałem logi przez formularz przysyłam informacje pomogą wiadomość odnaleźć
Przyjęliśmy zgłoszenie numer  "48633132" ref:!00D240aE05.!500Ti0LG7ce:ref

**Comment by PrzemyslawLukanowski** [comment]
Dziękuję, otrzymaliśmy Twoje dane. ClientId i clientSecret wyglądają na zakodowane prawidłowo, refresh token także jest poprawny, dlatego przekażę temat do dalszej weryfikacji. 

Problem może leżeć po stronie samego skryptu, który wykonuje tę operację. Możesz zasugerować się kodem widocznym w poniższych wątkach:

https://github.com/allegro/allegro-api/issues/10353#issuecomment-2493577009
https://github.com/allegro/allegro-api/issues/9791#issue-2500167866
https://github.com/allegro/allegro-api/issues/3374#issuecomment-662309716

być może uda Ci się dopatrzeć różnicę w Twoim kodzie, która może mieć wpływ na błędne przetwarzanie requestu.

Wrócę z odpowiedzią, gdy uda mi się otrzymać więcej informacji po analizie.

**Comment by PrzemyslawLukanowski** [comment]
Problem powoduje źle zakodowane client secret zaszyte w headerze. Jest w nim literówka (duże F zamiast małego f). 

Wysyłasz:

`...T2tGcXV...
`
powinno być:

`...T2tmcXV...`

**Comment by albertmazur** [comment]
Zmieniliśmy żeby generowało na  `...T2tmcXV...` zmieniając duże F na małe f ale dalej otrzymujemy ten sam błąd

**Comment by PrzemyslawLukanowski** [comment]
Z tego co widzę, nie był to jedyny fragment różniący się od tego, jak powinna wyglądać poprawna wartość. Wysyłałeś jeszcze:

`...lORnNO...`

a powinno być:

`...lOZnNO...`

Najlepiej, jeśli rozkodujesz swoją wartość w Basic, np. na [tej stronie](https://www.base64decode.org/) i porównasz, czy wartości zgadzają się z tymi, zarejestrowanymi na [stronie](https://apps.developer.allegro.pl/). Jeśli problem nadal będzie występować, poprosimy o nowy przykład z poprawionym Authorization.
